Streetcode/Streetcode.XUnitTest/MediatRTests/SourcesTests/GetCategoriesByStreetcodeIdTest.cs

```diff
@@ -0,0 +1,115 @@
+using Streetcode.DAL.Entities.Sources;
+using Streetcode.DAL.Entities.Streetcode.TextContent;
+using Streetcode.DAL.Repositories.Interfaces.Base;
+using Streetcode.WebApi.Controllers.Source;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Streetcode.XUnitTest.MediatRTests.SourcesTests
+{
+    public class GetCategoriesByStreetcodeIdTest
+    {
+        private readonly Mock<IRepositoryWrapper> _mockRepository;
+        private readonly Mock<IMapper> _mockMapper;
+        public GetCategoriesByStreetcodeIdTest()
+        {
+            _mockRepository = new Mock<IRepositoryWrapper>();
+            _mockMapper = new Mock<IMapper>();
+        }
+        [Fact]
+        public async Task ShouldReturnSuccesfully()
+        {
+
+            // arrange 
+            
+            _mockRepository.Setup(x => x.SourceCategoryRepository
+            .GetAllAsync(
+               It.IsAny<Expression<Func<SourceLinkCategory, bool>>>(),
+                It.IsAny<Func<IQueryable<SourceLinkCategory>,
+                IIncludableQueryable<SourceLinkCategory, object>>>()))
+            .ReturnsAsync(GetSourceLinkCategories());
+
+            _mockMapper.Setup(x => x.Map<IEnumerable<SourceLinkCategoryDTO>>(It.IsAny<IEnumerable<SourceLinkCategory>>()))
+                .Returns(GetSourceDTOs());
+
+            var handler = new GetCategoriesByStreetcodeIdHandler(_mockRepository.Object, _mockMapper.Object);
+            
+            // act
+            
+            var result = await handler.Handle(new GetCategoriesByStreetcodeIdQuery(1), CancellationToken.None);
+
+            // assert
+            Assert.Multiple(
+            () => Assert.NotNull(result),
+            () => Assert.IsType<List<SourceLinkCategoryDTO>>(result.ValueOrDefault)
+            );
+        }
+
+        [Theory]
+        [InlineData(1)]
+        public async Task ShouldReturnNull_NotExistingId(int id = 1)
+        {
+            // arrange 
+
+            _mockRepository.Setup(x => x.SourceCategoryRepository
+            .GetAllAsync(
+               It.IsAny<Expression<Func<SourceLinkCategory, bool>>>(),
+                It.IsAny<Func<IQueryable<SourceLinkCategory>,
+                IIncludableQueryable<SourceLinkCategory, object>>>()))
+            .ReturnsAsync(GetSourceLinkCategoriesNotExists());
+
+            _mockMapper.Setup(x => x.Map<IEnumerable<SourceLinkCategoryDTO>>(It.IsAny<IEnumerable<SourceLinkCategory>>()))
+                .Returns(GetSourceDTOs());
+
+            var handler = new GetCategoriesByStreetcodeIdHandler(_mockRepository.Object, _mockMapper.Object);
+
+            var expectedError = $"Cant find any source category with the streetcode id {id}";
+            // act
+
+            var result = await handler.Handle(new GetCategoriesByStreetcodeIdQuery(1), CancellationToken.None);
```

Comment:
it is a better practice to keep values in (test) variables - avoid using "magical" numbers